src/AlgebraicGeometry/Schemes/Covering/Objects/Attributes.jl

```diff
@@ -107,18 +107,22 @@ function inherit_decomposition_info!(
 
   for V in patches(orig_cov)
     # Collect the patches in `ref_cov` refining V
-    V_ref = [U for U in patches(ref_cov) if decomp_dict[U][1] === V]
-    # Collect the corresponding complement equations
-    comp_eqns = [decomp_dict[U][2] for U in V_ref]
+    V_ref = [(U, (UV, h)) for (U, (UV, h)) in decomp_dict if UV === V]
+    _compl(a) = sum(length(lifted_numerator(b)) + length(lifted_denominator(b)) for b in a[2][2]; init=0)
```

Comment:
A minor and purely optional nitpick (feel free to ignore):

So `UV` is always `V`. So why not do this?
```suggestion
    V_ref = [(U, h) for (U, (UV, h)) in decomp_dict if UV === V]
    _compl(a) = sum(length(lifted_numerator(b)) + length(lifted_denominator(b)) for b in a[2]; init=0)
```
and then the `for` loop in line 115 can be slightly simplified, and also line 120.


This avoids a little bit of storage space for those tuples.

Anyway, perhaps you have a reason for having it as it is (e.g. plans for further changes, keeping the code similar to something else etc.) -- so as I said, feel free to ignore.

Reply:
This makes sense. I just cluttered it together so that it would work for the moment. Thanks for the close look.